### ISSUE TITLE
Calling is_circular_planar() does not work on ZOO graphs

### ISSUE THREAD

**Issue by jaanos** [DiscreteZOO/DiscreteZOO-sage#3]
The method `is_circular_planar()` fails as it makes an immutable copy of an immutable graph and then tries to add a vertex to it, which does not work. This is s Sage issue which affects all immutable graphs, so a ticket should be opened on trac.

As of 5a623fe, the graphs are immutable, since it does not make sense to have a mutable graph with cached properties. The `copy` method is thus overloaded to return a `Graph` object if a mutable copy is requested.

A workaround is to use the `copy` function, which creates an immutable copy by calling the `copy(immutable = False)` method:

``` sage
sage: G = graphzoo.cvt.CVTGraph(10, 3)
sage: copy(G).is_circular_planar()
False
sage: G.is_circular_planar()
ValueError: Thou shalt not add a vertex to an immutable graph
```

**Comment by jaanos** [comment]
Here's the complete list of functions from the `GenericGraph` and `Graph` classes that do not work on immutable graphs:
- `GenericGraph.disjoint_routed_paths`
- `GenericGraph.genus`
- `GenericGraph.is_circular_planar`
- `GenericGraph.is_cut_edge`
- `GenericGraph.layout_planar`
- `GenericGraph.multicommodity_flow`
- `GenericGraph.set_planar_positions`
- `GenericGraph.steiner_tree`
- `GenericGraph.to_simple`
- `GenericGraph.vertex_disjoint_paths`
- `Graph.gomory_hu_tree`
- `Graph.is_long_antihole_free`
- `Graph.is_long_hole_free`
- `Graph.is_weakly_chordal`
- `Graph.join`
- `Graph.seidel_switching`
- `Graph.spanning_trees`
- `Graph.topological_minor`
- `Graph.tutte_polynomial`

**Comment by jaanos** [comment]
I have opened [ticket #19526](http://trac.sagemath.org/ticket/19526) on the Sage trac with the patch. Hopefully, this should go into Sage 6.10. I will close the issue when a version of Sage with the patch is released.

**Comment by jaanos** [comment]
This has been merged into Sage 6.10, so I'm closing the issue.
